### ISSUE TITLE
Samsung S6 6.0 Energy options break VPN

### ISSUE THREAD

**Issue by schwabe** [schwabe/ics-openvpn#471]
_sigh_

**Comment by MMD70** [comment]
Yes, i just trying to help someone on Dutch Synology forum.
Android 6.0.1 Samsung S6
Error: "UDP Send error: send : Operation not permitted."
Tried to play with txqueuelen but no succes.

So it`s the energy options?

**Comment by lomperd** [comment]
![1](https://cloud.githubusercontent.com/assets/15231579/14217400/0cf8b08a-f84e-11e5-825b-4a8e19284045.png)
![2](https://cloud.githubusercontent.com/assets/15231579/14217402/11129abe-f84e-11e5-86df-153991c7f472.png)
![3](https://cloud.githubusercontent.com/assets/15231579/14217403/1441dd4e-f84e-11e5-8550-26f5bcaabd12.png)

This is the solution.

Thnx for the help @MMD70 

Edit : Worked only for one time , unfortunately , now the same message again ...

**Comment by MMD70** [comment]
Ok, the user found a solution after seeing this issue here.
Settings-->Battery-->Details
Then disable the energy saving for the App.

**Comment by schwabe** [comment]
@MMD70 yes.

documented by commit 2138a6ad52b7b48088eb8b46b46b86c2994530f7

**Comment by MMD70** [comment]
Ok thank you,
he found out yesterday, excempting from Powersave features solves it.

**Comment by vinodkutty** [comment]
I don't know what exactly "exempting from Powersave features" means, but I experienced this issue on a Galaxy S7 Edge device running Android 6.0.1 with Power Saving Mode enabled and believe it's more generally applicable to similar devices with this option. The other mode is "Ultra Power Saving", which is designed to make the phone just a phone and thus not expected to allow OpenVPN to work at all [skipped nostalgic remark about flip phones]

These are the settings I found:
- Settings -> Battery -> App power saving -> DETAIL -> OpenVPN Connect -> select "Turned Off"
  This seems to be mainly for deciding what to do with apps that haven't been used in more than 3 days.
- Settings -> Battery -> Power Saving Mode -> "Restrict all background data": Off
  This seems to be the most relevant for this issue, and affects other background data services (such as mail sync). Setting this "Off" fixed the error.

I expect the more coarse-grained method is to simply turning off "Power Saving Mode" as a whole, but I haven't verified this.
